### PR TITLE
AWS docs

### PR DESCRIPTION
**Pull request by frostebite**
Added page Remote builder under GitHub

### CODE REVIEW

File: docs/github/v2/06-remote-builder.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Remote builder
+
+## What is it?
+
+GitHub Actions run on build machines provided by GitHub. You can see info about the specifications of the build machines [here]
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:5]:
```suggestion
GitHub Actions by default run on build machines provided by GitHub. You can see info about the specifications of the build machines [here]
```

File: docs/github/v2/06-remote-builder.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+(https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners).
+
+For Unity projects the disc size is quite small. You may experience an error related to running out of disc space. You may also want to run the build on a server with more memory or CPU resources.
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:8]:
```suggestion
For Unity projects the available disk size is quite small. You may experience an error related to running out of disk space. You may also want to run the build on a server with more memory or processing resources.
```

Source for [disk](https://www.oxfordlearnersdictionaries.com/definition/english/disc#:~:text=The%20usual%20spelling%20is%20disc,even%20in%20(British%20English).)

File: docs/github/v2/06-remote-builder.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+**You can use Remote Builder to send a build from any supported git platform to any of the supported remote builder types.**
+
+## Key Features
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:12]:
```suggestion
## Key features
```

File: docs/github/v2/06-remote-builder.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
+# Remote builder
+
+## What is it?
+
+GitHub Actions run on build machines provided by GitHub. You can see info about the specifications of the build machines [here]
+(https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners).
+
+For Unity projects the disc size is quite small. You may experience an error related to running out of disc space. You may also want to run the build on a server with more memory or CPU resources.
+
+**You can use Remote Builder to send a build from any supported git platform to any of the supported remote builder types.**
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:10]:
```suggestion
**You can use remote builder to send a build from any supported git platform to any of the supported remote builder types.**
```

Either this is a keyword (in which case we can use caps) or its lower case. Looking at the title and no introduction of the term we do not intend a special keyword here right?

File: docs/github/v2/06-remote-builder.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+## Key Features
+
+- **Run builds on a cloud provider like AWS.** _This is billed on demand and costs nothing except unless you store the builds (which is cheap) while no builds are running._
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:14]:
```suggestion
- **Run builds on a cloud provider like AWS.** _This is billed on demand and costs nothing unless you store the builds._
```

File: docs/github/v2/06-remote-builder.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+✔️ **AWS** is supported.
+
+❌ **GCP** planned, Cloud Run needs to release support for NFS storage.
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:18]:
```suggestion
⚠ **GCP** planned, Google Cloud Run needs to release support for NFS.
```

File: docs/github/v2/06-remote-builder.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+❌ **Azure** planned, already supported.
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:20]:
```suggestion
⚠ **Azure** planned, the needed technologies are already supported.
```

File: docs/github/v2/06-remote-builder.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+- **Run builds on a Kubernetes cluster.** _Usually the cluster needs to be up and running at all times at all times as starting up a cluster is slow. Using Google Cloud's Kubernetes Autopilot you can scale down to the free tier automatically while not in use._
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:22]:
```suggestion
- **Run builds on a Kubernetes cluster.** _Usually the cluster needs to be up and running at all times, as starting up a cluster is slow. Using Google Cloud's Kubernetes Autopilot you can scale down to the free tier automatically while not in use._
```

File: docs/github/v2/06-remote-builder.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+✔️ All Kubernetes providers that support persistent volumes are supported.
+
+### When to use GitHub self-hosted runner or Remote Builder
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:26]:
```suggestion
### When to use GitHub self-hosted runner or remote builder?
```

Comment (webbertakken) [reply docs/github/v2/06-remote-builder.md]: Gonna stop here correcting all `remote builder` and `Remote Builder`. Please pick one and replace it throughout :)

File: docs/github/v2/06-remote-builder.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+[GitHub self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) and Remote Builder can enable you to build larger projects.
+
+Self-hosted runners are best used when you already have a server available, running 24/7 that you can setup as a runner. And you're happy to maintain and keep that server available and running.
+
+Remote Builder is better if you don't have a server setup or don't want to manage or maintain a build server.
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:32]:
```suggestion
Remote Builder is better if you don't have a server setup or don't want to manage or maintain your own build server.
```

File: docs/github/v2/06-remote-builder.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+## Basic Remote Builder setup with AWS
+
+### Requirements
+
+- You must have an AWS account setup and ready to create resources.
+- It is suggested you have a workflow setup from the [builder section](builder).
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:39]: This doesn't link anywhere yet

Comment (frostebite) [reply docs/github/v2/06-remote-builder.md:39]: relative link??

File: docs/github/v2/06-remote-builder.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+### AWS Credentials
+
+Setup the following as `env` variables for the GitHub build step:
+
+- `AWS_ACCESS_KEY_ID`
+- `AWS_SECRET_ACCESS_KEY`
+- `AWS_DEFAULT_REGION` (should be the same AWS region as the base stack e.g `eu-west-2`)
+
+_Note:_
+_This enables us to access AWS resources from the GitHub builder._
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:63]:
```suggestion
_This enables us to access AWS resources from Unity Builder._
```

File: docs/github/v2/06-remote-builder.md (hunk 11 above)
Comment (frostebite) [reply docs/github/v2/06-remote-builder.md:39]: And in the vercel preview env

File: docs/github/v2/06-remote-builder.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+
+### GitHub Access Token
+
+Include the following `with` variable for the GitHub build step:
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:67]:
```suggestion
Include the following variable in the `with` section of the builder step:
```

File: docs/github/v2/06-remote-builder.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+Add the following parameters to the build step described in the [builder section](builder):
+
+- `remoteBuildCluster` (should be `aws`)
+- `awsStackName` (should be the name you gave the base stack in aws)
+- `remoteBuildMemory` (the memory the build container should be given)
+- `remoteBuildCpu` (the CPU units the build container should be given)
+
+#### Allowed CPU/Memory combinations
+
+There are some limitations to the CPU and Memory parameters. AWS will only accept the following combinations:
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:85]: Is there a link to these limitations somewhere?

File: docs/github/v2/06-remote-builder.md [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+
+```yaml
+- remoteBuildMemory: 2
+- remoteBuildCpu: 0.5
+```
+
+This information may go out of date if AWS update this. Please find the latest documentation at the reference link below.
+
+_Reference:_
+[AWS Fargate Documentation, Task Definitions, Task CPU and memory](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html#fargate-task-defs)
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:102]: Recommend just linking this under the word `[limitations]` above.

File: docs/github/v2/06-remote-builder.md (hunk 14 above)
Comment (frostebite) [reply docs/github/v2/06-remote-builder.md:84]: This is just below :)

File: docs/github/v2/06-remote-builder.md (hunk 15 above)
Comment (frostebite) [reply docs/github/v2/06-remote-builder.md:102]: There isn't an anchor to the *exact* bit we want to link to. So I wanted to give the full path nice and clearly. Any suggestions?

File: docs/github/v2/06-remote-builder.md [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+    AWS_DEFAULT_REGION: eu-west-2
+  with:
+    remoteBuildCluster: aws
+    projectPath: ${{ matrix.projectPath }}
+    unityVersion: ${{ matrix.unityVersion }}
+    targetPlatform: ${{ matrix.targetPlatform }}
+    githubToken: ${{ secrets.GITHUB_TOKEN }}
+```
+
+A full workflow example can be seen in our [own CI workflow](https://github.com/game-ci/unity-builder/blob/main/.github/workflows/aws-tests.yml).
```
Comment (webbertakken) [inline docs/github/v2/06-remote-builder.md:122]:
```suggestion
A full workflow example can be seen in builder's [test workflow](https://github.com/game-ci/unity-builder/blob/main/.github/workflows/aws-tests.yml).
```

**Review by webbertakken — APPROVED** [review]

**Comment by webbertakken** [comment]
LGTM. Nice work @frostebite!

**Pull request merged**